added `saveFile` action

mehsaandev opened the pull request.

kmahmood74 commented:
@mehsaandev no reviewer? is this draft?

mehsaandev commented:
Yes this is draft, finishing it.

snehmehta commented on modules/ensemble/lib/action/save_file.dart:
```diff
@@ -0,0 +1,190 @@
+        }
+      } else if (source != null) {
+        // If blobData is not available, check for source (network URL)
+        Dio dio = Dio();
```
why use dio when we have http package for network handling?

snehmehta commented on modules/ensemble/lib/action/save_file.dart:
```diff
@@ -0,0 +1,190 @@
+
+  /// Notify the Android media store about the new file
+  Future<void> _notifyFileSystem(String filePath) async {
+    try {
```
Running a process is not recommended if possible please find another way

snehmehta commented on modules/ensemble/lib/action/save_file.dart:
```diff
@@ -0,0 +1,190 @@
+
+  Future<void> _saveImageToDCIM(String fileName, Uint8List fileBytes) async {
+    try {
+      final result = await ImageGallerySaver.saveImage(
```
what is reason for using ImageGallerySaver, cause adding in android would be placing file on that location. is it ios?

mehsaandev replied:
If I just place the file in that location it doesn't register the image in the gallery, using ImageGallerySaver it automatically registers the image to the gallery in the respective platforms I.e. Android and IOS

kmahmood74 commented:
Due to the urgency of this fix, I would suggest having a meeting to go
through the PR. We need these done today as the customer is waiting

On Mon, Jan 6, 2025 at 4:49 AM Muhammad Ehsaan ***@***.***>
wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In modules/ensemble/lib/action/save_file.dart
> <https://github.com/EnsembleUI/ensemble/pull/1799#discussion_r1904113738>:
>
> > +
> +      if (type == 'image') {
> +        // Save images to Default Image Path
> +        await _saveImageToDCIM(fileName!, fileBytes);
> +      } else if (type == 'document') {
> +        // Save documents to Documents folder
> +        await _saveDocumentToDocumentsFolder(fileName!, fileBytes);
> +      }
> +    } catch (e) {
> +      throw Exception('Failed to save file: $e');
> +    }
> +  }
> +
> +  Future<void> _saveImageToDCIM(String fileName, Uint8List fileBytes) async {
> +    try {
> +      final result = await ImageGallerySaver.saveImage(
>
> If I just place the file in that location it doesn't register the image in
> the gallery, using ImageGallerySaver it automatically registers the image
> to the gallery in the respective platforms I.e. Android and IOS
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/EnsembleUI/ensemble/pull/1799#discussion_r1904113738>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACMZUAVTSRUKVKFFYWXNST32JJ3W3AVCNFSM6AAAAABUN44GL6VHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDKMZSGA3TMMRRG4>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>

mehsaandev commented:
I've made the required changes. 
- For `ImageGallerySaver` It registers the file to gallery so that's I added this, only saving the file to Pictures path won't register it to gallery
- Removed Document registry, saving the document is sufficient.
- Replaced Dio with HTTP

@snehmehta PTAL
